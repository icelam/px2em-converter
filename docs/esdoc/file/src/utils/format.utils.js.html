<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <base data-ice="baseUrl" href="../../../">
  <title data-ice="title">src/utils/format.utils.js | PX2Em Converter</title>
  <link type="text/css" rel="stylesheet" href="css/style.css">
  <link type="text/css" rel="stylesheet" href="css/prettify-tomorrow.css">
  <script src="script/prettify/prettify.js"></script>
  <script src="script/manual.js"></script>
<meta name="description" content="PX2EM is a conversion tool which allows you to calculate the em sizes from pixels. Just add the base font size and the pixels to convert, and you&apos;re done!"><meta property="twitter:card" content="summary"><meta property="twitter:title" content="PX2Em Converter"><meta property="twitter:description" content="PX2EM is a conversion tool which allows you to calculate the em sizes from pixels. Just add the base font size and the pixels to convert, and you&apos;re done!"><meta property="twitter:image" content="./preview.png"></head>
<body class="layout-container" data-ice="rootContainer">

<header>
  <a href="./" style="display: flex; align-items: center;"><img src="./image/brand_logo.png" style="width:34px;"></a>
  
  <a href="identifiers.html">Reference</a>
  <a href="source.html">Source</a>
  <a href="test.html" data-ice="testLink">Test</a>
  <div class="search-box">
  <span>
    <img src="./image/search.png">
    <span class="search-input-edge"></span><input class="search-input"><span class="search-input-edge"></span>
  </span>
    <ul class="search-result"></ul>
  </div>
<a href="http://github.com/icelam/px2em-converter">Repository</a></header>

<nav class="navigation" data-ice="nav"><div>
  <ul>
    
  <li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#hooks">hooks</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-useDebounce">useDebounce</a></span></span></li>
<li data-ice="doc"><a data-ice="dirPath" class="nav-dir-path" href="identifiers.html#utils">utils</a><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-convertHexToRgb">convertHexToRgb</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-convertPixelRangeToArray">convertPixelRangeToArray</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-deduplicateArray">deduplicateArray</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-removeExtraSpacesAndCommas">removeExtraSpacesAndCommas</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-roundToNearest">roundToNearest</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-sortArrayAscending">sortArrayAscending</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-clearAll">clearAll</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-getKey">getKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-removeKey">removeKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-setKey">setKey</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-isValidFontSize">isValidFontSize</a></span></span></li>
<li data-ice="doc"><span data-ice="kind" class="kind-function">F</span><span data-ice="name"><span><a href="function/index.html#static-function-isValidPositiveInteger">isValidPositiveInteger</a></span></span></li>
</ul>
</div>
</nav>

<div class="content" data-ice="content"><h1 data-ice="title">src/utils/format.utils.js</h1>
<pre class="source-code line-number raw-source-code"><code class="prettyprint linenums" data-ice="content">import { isValidPositiveInteger } from &apos;./validate.utils&apos;;

/**
 * Coverts any HEX color code to RGB
 * @param {number} value - Interger / floating point value that needs to be rounded
 * @param {number} precision - Number of decimal places
 * @returns {number} Rounded floating point value
 */
export const roundToNearest = (value, precision) =&gt; {
  if (!Number.isInteger(precision)) {
    return value;
  }
  // parseFloat for trimming trailing zeros
  return parseFloat(value.toFixed(precision));
};

/**
 * Deduplicates any array
 * @param {array} arr - Array to be deduplicated
 * @returns {array} Deduplicated array If `arr` provided is not an array, returns an empty array
 */
export const deduplicateArray = (arr) =&gt; {
  if (!Array.isArray(arr)) {
    return [];
  }

  const arraySet = new Set(arr);
  return Array.from(arraySet);
};

/**
 * Remove multiple consecutive spaces, trialing space and comma
 * @param {string} str - String to process
 * @returns {string} String which its multiple consecutive spaces,
 * trialing space and comma are removed
 */
export const removeExtraSpacesAndCommas = (str) =&gt; str
  .replace(/\s+/g, &apos; &apos;)
  .replace(/\s$/, &apos;&apos;)
  .replace(/,$/, &apos;&apos;);

/**
 * Sort array in ascending order
 * @param {array} arr - Array to sort
 * @returns {array} Array sorted in ascending order
 */
const sortArrayAscending = (arr) =&gt; {
  const newArray = [...arr];
  return newArray.sort((a, b) =&gt; a - b);
};

/**
 * Convert pixel range to list of pixels
 * @param {string} str - Pixel range, e.g. `&quot;12 - 20, 50, 100&quot;`
 * @returns {array} List of pixels, e.g. `[12, 13, 14, 15, 16, 17, 18, 19, 20, 50, 100]``
 */
export const convertPixelRangeToArray = (str) =&gt; {
  try {
    const pixels = [];

    if (!str) {
      throw new Error(&apos;EMPTY_INPUT&apos;);
    }

    const formattedString = removeExtraSpacesAndCommas(str);
    const rangeItems = formattedString.split(&apos;,&apos;);

    rangeItems.forEach((item) =&gt; {
      let currentDigit = &apos;&apos;;
      const operations = [];

      const pushDigit = () =&gt; {
        operations.push(currentDigit);
        currentDigit = &apos;&apos;;
      };

      /**
       * Parse string to operations:
       *  - &quot;11&quot; to [&quot;11&quot;]
       *  - &quot;11 - 13&quot; to [&quot;11&quot;, &quot;-&quot;, &quot;13&quot;]
       *  - &quot;1-3&quot; to [&quot;1&quot;, &quot;-&quot;, &quot;1&quot;]
       */
      for (let i = 0; i &lt; item.length; i++) {
        if (item[i] === &apos; &apos;) {
          currentDigit &amp;&amp; pushDigit();
        } else if (isValidPositiveInteger(item[i])) {
          currentDigit += item[i];
        } else if (item[i] === &apos;-&apos;) {
          currentDigit &amp;&amp; pushDigit();
          operations.push(&apos;-&apos;);
        } else {
          throw new Error(&apos;INVALID_INPUT&apos;);
        }
      }

      currentDigit &amp;&amp; pushDigit();

      /**
       * Convert operations to pixels:
       *  - Valid Case 1: Single positive integer
       *  - Valid Case 2: Pixel range (e.g. 1-10)
       */
      if (operations.length !== 1 &amp;&amp; operations.length !== 3) {
        throw new Error(&apos;INVALID_OPERATION_LENGTH&apos;);
      }

      if (operations.length === 1) {
        if (isValidPositiveInteger(operations[0])) {
          pixels.push(parseInt(operations[0], 10));
        } else {
          throw new Error(&apos;INVALID_NUMBER&apos;);
        }
      } else {
        const [leftString, operator, rightString] = operations;
        if (isValidPositiveInteger(leftString) &amp;&amp; operator === &apos;-&apos; &amp;&amp; isValidPositiveInteger(rightString)) {
          const leftNumber = parseInt(leftString, 10);
          const rightNumber = parseInt(rightString, 10);
          if (leftNumber &lt;= rightNumber) {
            for (let i = leftNumber; i &lt;= rightNumber; i++) {
              pixels.push(parseInt(i, 10));
            }
          } else {
            throw new Error(&apos;INVALID_RANGE&apos;);
          }
        } else {
          throw new Error(&apos;INVALID_OPERATION&apos;);
        }
      }
    });

    const uniquePixels = deduplicateArray(pixels);
    const sortedPixels = sortArrayAscending(uniquePixels);
    return sortedPixels;
  } catch (error) {
    return [];
  }
};
</code></pre>

</div>

<footer class="footer">
  Generated by <a href="https://esdoc.org">ESDoc<span data-ice="esdocVersion">(1.1.0)</span><img src="./image/esdoc-logo-mini-black.png"></a>
</footer>

<script src="script/search_index.js"></script>
<script src="script/search.js"></script>
<script src="script/pretty-print.js"></script>
<script src="script/inherited-summary.js"></script>
<script src="script/test-summary.js"></script>
<script src="script/inner-link.js"></script>
<script src="script/patch-for-local.js"></script>
</body>
</html>
