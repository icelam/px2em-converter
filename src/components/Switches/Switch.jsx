import React, { memo } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const SwitchWrapper = styled.div`
  margin: 0.25rem 0;
`;

const SwitchLabel = styled.span`
  font-size: 0.875rem;
  color: ${({ theme: { color } }) => color.secondaryTextColor};
  margin: 0 0.3125rem 0 0;
`;

const SwitchButton = styled.button`
  display: inline-block;
  border: 0.0625rem solid ${({ theme: { color } }) => color.switchButtonBorder};
  border-radius: 0;
  box-shadow: none;
  background-color: ${({ selected, theme: { color } }) => (
    selected
      ? color.switchButtonActiveBackgroundColor
      : color.switchButtonBackgroundColor
  )};
  color: ${({ selected, theme: { color } }) => (
    selected
      ? color.switchButtonActiveTextColor
      : color.switchButtonTextColor
  )};
  padding: 0.3125rem 0.875rem;
  margin: 0 0 0 -0.0625rem;
  font-size: 0.875rem;
  font-family: inherit;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
  ${({ buttonWidth }) => buttonWidth && `min-width: ${buttonWidth};`}
`;

const Switch = ({
  label, value, onChange, options, buttonWidth
}) => {
  const handleChange = (selectedValue) => {
    onChange(selectedValue);
  };

  return (
    <SwitchWrapper className="switch">
      {label && <SwitchLabel className="switch__label">Unit:</SwitchLabel>}
      {
        options && Array.isArray(options) && options.map(({
          label: optionLabel, value: optionValue
        }) => (
          <SwitchButton
            className="switch__button"
            key={optionValue}
            selected={optionValue === value}
            onClick={() => { handleChange(optionValue); }}
            buttonWidth={buttonWidth}
          >
            {optionLabel}
          </SwitchButton>
        ))
      }
    </SwitchWrapper>
  );
};

Switch.propTypes = {
  label: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
    PropTypes.bool
  ]),
  onChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([
      PropTypes.number,
      PropTypes.string,
      PropTypes.bool
    ]).isRequired
  })).isRequired,
  buttonWidth: PropTypes.string
};

Switch.defaultProps = {
  label: undefined,
  value: '',
  buttonWidth: undefined
};

export default memo(Switch);
