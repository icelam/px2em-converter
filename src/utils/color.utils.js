const convertHexToRgb = (color) => {
  const colorCode = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(color.toLowerCase());

  return colorCode
    ? `${parseInt(colorCode[1], 16)}, ${parseInt(colorCode[2], 16)}, ${parseInt(colorCode[3], 16)}`
    : '';
};

export default convertHexToRgb;
