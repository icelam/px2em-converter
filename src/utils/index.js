export { default as convertHexToRgb } from './color.utils';
export * from './format.utils';
export { default as localStorageUtils } from './storage.utils';
export * from './validate.utils';
