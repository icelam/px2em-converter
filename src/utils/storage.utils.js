const setKey = (key, data) => {
  if (!key) { return; }
  window.localStorage.setItem(key, data);
};

const getKey = (key) => window.localStorage.getItem(key);

const removeKey = (key) => {
  window.localStorage.removeItem(key);
};

const clearAll = () => {
  window.localStorage.clear();
};

export default {
  setKey,
  getKey,
  removeKey,
  clearAll
};
