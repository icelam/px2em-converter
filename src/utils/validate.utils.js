/**
 * Validate if user input is a positive integer
 * @param {string} str - User input
 * @returns {boolean} `true` for positive integers, `false` for negative integers and other inputs
 */
export const isValidPositiveInteger = (str) => {
  const re = /^\d+$/;
  return re.test(str);
};

/**
 * Validate if user input is a valid font size
 * @param {string} str - User input
 * @returns {boolean} `true` for decimal and integer values, `false` for other inputs
 */
export const isValidFontSize = (str) => {
  const re = /^[0-9]+(\.[0-9]+)?$/;
  return re.test(str);
};
