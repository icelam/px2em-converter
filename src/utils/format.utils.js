import { isValidPositiveInteger } from './validate.utils';

export const roundToNearest = (value, precision) => {
  if (!Number.isInteger(precision)) {
    return value;
  }
  // parseFloat for trimming trailing zeros
  return parseFloat(value.toFixed(precision));
};

export const deduplicateArray = (arr) => {
  if (!Array.isArray(arr)) {
    return [];
  }

  const arraySet = new Set(arr);
  return Array.from(arraySet);
};

export const removeExtraSpacesAndCommas = (str) => str
  .replace(/\s+/g, ' ')
  .replace(/\s$/, '')
  .replace(/,$/, '');

const sortArrayAscending = (arr) => {
  const newArray = [...arr];
  return newArray.sort((a, b) => a - b);
};

export const convertPixelRangeToArray = (str) => {
  try {
    const pixels = [];

    if (!str) {
      throw new Error('EMPTY_INPUT');
    }

    const formattedString = removeExtraSpacesAndCommas(str);
    const rangeItems = formattedString.split(',');

    rangeItems.forEach((item) => {
      let currentDigit = '';
      const operations = [];

      const pushDigit = () => {
        operations.push(currentDigit);
        currentDigit = '';
      };

      /**
       * Parse string to operations:
       *  - "11" to ["11"]
       *  - "11 - 13" to ["11", "-", "13"]
       *  - "1-3" to ["1", "-", "1"]
       */
      for (let i = 0; i < item.length; i++) {
        if (item[i] === ' ') {
          currentDigit && pushDigit();
        } else if (isValidPositiveInteger(item[i])) {
          currentDigit += item[i];
        } else if (item[i] === '-') {
          currentDigit && pushDigit();
          operations.push('-');
        } else {
          throw new Error('INVALID_INPUT');
        }
      }

      currentDigit && pushDigit();

      /**
       * Convert operations to pixels:
       *  - Valid Case 1: Single positive integer
       *  - Valid Case 2: Pixel range (e.g. 1-10)
       */
      if (operations.length !== 1 && operations.length !== 3) {
        throw new Error('INVALID_OPERATION_LENGTH');
      }

      if (operations.length === 1) {
        if (isValidPositiveInteger(operations[0])) {
          pixels.push(parseInt(operations[0], 10));
        } else {
          throw new Error('INVALID_NUMBER');
        }
      } else {
        const [leftString, operator, rightString] = operations;
        if (isValidPositiveInteger(leftString) && operator === '-' && isValidPositiveInteger(rightString)) {
          const leftNumber = parseInt(leftString, 10);
          const rightNumber = parseInt(rightString, 10);
          if (leftNumber <= rightNumber) {
            for (let i = leftNumber; i <= rightNumber; i++) {
              pixels.push(parseInt(i, 10));
            }
          } else {
            throw new Error('INVALID_RANGE');
          }
        } else {
          throw new Error('INVALID_OPERATION');
        }
      }
    });

    const uniquePixels = deduplicateArray(pixels);
    const sortedPixels = sortArrayAscending(uniquePixels);
    return sortedPixels;
  } catch (error) {
    return [];
  }
};
