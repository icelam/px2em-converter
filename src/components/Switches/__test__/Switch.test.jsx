import React from 'react';
import { themes } from '@styles';
import { mountWithProvider } from '../../../../jest.utils';
import Switch from '../Switch';

const options1 = [
  { label: 'em', value: 'em' },
  { label: 'rem', value: 'rem' }
];

const options2 = [
  { label: '1', value: 1 },
  { label: '2', value: 2 },
  { label: '3', value: 3 },
  { label: '4', value: 4 }
];

const onChange = jest.fn();

afterEach(() => {
  onChange.mockClear();
});

describe('Switch', () => {
  it('should render', () => {
    const wrapper = mountWithProvider(<Switch onChange={onChange} value="em" options={options1} label="Unit: " />);
    expect(wrapper.find('div.switch')).toHaveLength(1);
    expect(wrapper.find('div.switch span.switch__label')).toHaveLength(1);
    expect(wrapper.find('div.switch button.switch__button')).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
  });

  it('should display switch with label and 2 options', () => {
    const label = 'Unit: ';
    const wrapper = mountWithProvider(
      <Switch onChange={onChange} options={options1} label={label} />
    );
    expect(wrapper.find('div.switch span.switch__label')).toHaveLength(1);
    expect(wrapper.find('div.switch span.switch__label').text()).toEqual(label.trim());
    expect(wrapper.find('div.switch button.switch__button')).toHaveLength(2);
    expect(wrapper.find('div.switch button.switch__button').at(0).text()).toEqual(options1[0].label);
    expect(wrapper.find('div.switch button.switch__button').at(1).text()).toEqual(options1[1].label);
  });

  it('should display switch without label and 4 options', () => {
    const wrapper = mountWithProvider(<Switch onChange={onChange} options={options2} />);
    expect(wrapper.find('div.switch span.switch__label')).toHaveLength(0);
    expect(wrapper.find('div.switch button.switch__button')).toHaveLength(4);
    expect(wrapper.find('div.switch button.switch__button').at(0).text()).toEqual(options2[0].label);
    expect(wrapper.find('div.switch button.switch__button').at(1).text()).toEqual(options2[1].label);
    expect(wrapper.find('div.switch button.switch__button').at(2).text()).toEqual(options2[2].label);
    expect(wrapper.find('div.switch button.switch__button').at(3).text()).toEqual(options2[3].label);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to highlight selected value', () => {
    const wrapper = mountWithProvider(<Switch onChange={onChange} value="rem" options={options1} />);
    const {
      dark: {
        color: {
          switchButtonBackgroundColor,
          switchButtonTextColor,
          switchButtonActiveBackgroundColor,
          switchButtonActiveTextColor
        }
      }
    } = themes;
    expect(wrapper.find('div.switch button.switch__button').at(0))
      .toHaveStyleRule('background-color', switchButtonBackgroundColor);
    expect(wrapper.find('div.switch button.switch__button').at(0))
      .toHaveStyleRule('color', switchButtonTextColor);
    expect(wrapper.find('div.switch button.switch__button').at(1))
      .toHaveStyleRule('background-color', switchButtonActiveBackgroundColor);
    expect(wrapper.find('div.switch button.switch__button').at(1))
      .toHaveStyleRule('color', switchButtonActiveTextColor);
    expect(wrapper).toMatchSnapshot();
  });

  it('should be able to define minimum button width', () => {
    const wrapper = mountWithProvider(<Switch onChange={onChange} options={options1} buttonWidth="2.5rem" />);
    expect(wrapper.find('div.switch button.switch__button').at(0)).toHaveStyleRule('min-width', '2.5rem');
    expect(wrapper.find('div.switch button.switch__button').at(1)).toHaveStyleRule('min-width', '2.5rem');
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger onChange when clicking on switch buttons', () => {
    const wrapper = mountWithProvider(<Switch onChange={onChange} value="rem" options={options1} />);
    wrapper.find('div.switch button.switch__button').at(0).simulate('click');
    expect(onChange).toHaveBeenCalledTimes(1);
    wrapper.find('div.switch button.switch__button').at(1).simulate('click');
    expect(onChange).toHaveBeenCalledTimes(2);
  });
});
