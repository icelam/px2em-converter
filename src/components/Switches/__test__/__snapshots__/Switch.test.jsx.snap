// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should be able to define minimum button width 1`] = `
.c0 {
  margin: 0.25rem 0;
}

.c1 {
  display: inline-block;
  border: 0.0625rem solid #004b4d;
  border-radius: 0;
  box-shadow: none;
  background-color: #05171d;
  color: #828b8e;
  padding: 0.3125rem 0.875rem;
  margin: 0 0 0 -0.0625rem;
  font-size: 0.875rem;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
  min-width: 2.5rem;
}

<Switch
  buttonWidth="2.5rem"
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "em",
        "value": "em",
      },
      Object {
        "label": "rem",
        "value": "rem",
      },
    ]
  }
>
  <styled.div
    className="switch"
  >
    <div
      className="c0 switch"
    >
      <styled.button
        buttonWidth="2.5rem"
        className="switch__button"
        key="em"
        onClick={[Function]}
        selected={false}
      >
        <button
          className="c1 switch__button"
          onClick={[Function]}
          selected={false}
        >
          em
        </button>
      </styled.button>
      <styled.button
        buttonWidth="2.5rem"
        className="switch__button"
        key="rem"
        onClick={[Function]}
        selected={false}
      >
        <button
          className="c1 switch__button"
          onClick={[Function]}
          selected={false}
        >
          rem
        </button>
      </styled.button>
    </div>
  </styled.div>
</Switch>
`;

exports[`Switch should be able to highlight selected value 1`] = `
.c0 {
  margin: 0.25rem 0;
}

.c1 {
  display: inline-block;
  border: 0.0625rem solid #004b4d;
  border-radius: 0;
  box-shadow: none;
  background-color: #05171d;
  color: #828b8e;
  padding: 0.3125rem 0.875rem;
  margin: 0 0 0 -0.0625rem;
  font-size: 0.875rem;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
}

.c2 {
  display: inline-block;
  border: 0.0625rem solid #004b4d;
  border-radius: 0;
  box-shadow: none;
  background-color: #004b4d;
  color: #ffffff;
  padding: 0.3125rem 0.875rem;
  margin: 0 0 0 -0.0625rem;
  font-size: 0.875rem;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
}

<Switch
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "em",
        "value": "em",
      },
      Object {
        "label": "rem",
        "value": "rem",
      },
    ]
  }
  value="rem"
>
  <styled.div
    className="switch"
  >
    <div
      className="c0 switch"
    >
      <styled.button
        className="switch__button"
        key="em"
        onClick={[Function]}
        selected={false}
      >
        <button
          className="c1 switch__button"
          onClick={[Function]}
          selected={false}
        >
          em
        </button>
      </styled.button>
      <styled.button
        className="switch__button"
        key="rem"
        onClick={[Function]}
        selected={true}
      >
        <button
          className="c2 switch__button"
          onClick={[Function]}
          selected={true}
        >
          rem
        </button>
      </styled.button>
    </div>
  </styled.div>
</Switch>
`;

exports[`Switch should display switch without label and 4 options 1`] = `
.c0 {
  margin: 0.25rem 0;
}

.c1 {
  display: inline-block;
  border: 0.0625rem solid #004b4d;
  border-radius: 0;
  box-shadow: none;
  background-color: #05171d;
  color: #828b8e;
  padding: 0.3125rem 0.875rem;
  margin: 0 0 0 -0.0625rem;
  font-size: 0.875rem;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
}

<Switch
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "1",
        "value": 1,
      },
      Object {
        "label": "2",
        "value": 2,
      },
      Object {
        "label": "3",
        "value": 3,
      },
      Object {
        "label": "4",
        "value": 4,
      },
    ]
  }
>
  <styled.div
    className="switch"
  >
    <div
      className="c0 switch"
    >
      <styled.button
        className="switch__button"
        key="1"
        onClick={[Function]}
        selected={false}
      >
        <button
          className="c1 switch__button"
          onClick={[Function]}
          selected={false}
        >
          1
        </button>
      </styled.button>
      <styled.button
        className="switch__button"
        key="2"
        onClick={[Function]}
        selected={false}
      >
        <button
          className="c1 switch__button"
          onClick={[Function]}
          selected={false}
        >
          2
        </button>
      </styled.button>
      <styled.button
        className="switch__button"
        key="3"
        onClick={[Function]}
        selected={false}
      >
        <button
          className="c1 switch__button"
          onClick={[Function]}
          selected={false}
        >
          3
        </button>
      </styled.button>
      <styled.button
        className="switch__button"
        key="4"
        onClick={[Function]}
        selected={false}
      >
        <button
          className="c1 switch__button"
          onClick={[Function]}
          selected={false}
        >
          4
        </button>
      </styled.button>
    </div>
  </styled.div>
</Switch>
`;

exports[`Switch should render 1`] = `
.c0 {
  margin: 0.25rem 0;
}

.c1 {
  font-size: 0.875rem;
  font-weight: 600;
  color: #828b8e;
  margin: 0 0.3125rem 0 0;
}

.c2 {
  display: inline-block;
  border: 0.0625rem solid #004b4d;
  border-radius: 0;
  box-shadow: none;
  background-color: #004b4d;
  color: #ffffff;
  padding: 0.3125rem 0.875rem;
  margin: 0 0 0 -0.0625rem;
  font-size: 0.875rem;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
}

.c3 {
  display: inline-block;
  border: 0.0625rem solid #004b4d;
  border-radius: 0;
  box-shadow: none;
  background-color: #05171d;
  color: #828b8e;
  padding: 0.3125rem 0.875rem;
  margin: 0 0 0 -0.0625rem;
  font-size: 0.875rem;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  outline: none;
}

<Switch
  label="Unit: "
  onChange={[MockFunction]}
  options={
    Array [
      Object {
        "label": "em",
        "value": "em",
      },
      Object {
        "label": "rem",
        "value": "rem",
      },
    ]
  }
  value="em"
>
  <styled.div
    className="switch"
  >
    <div
      className="c0 switch"
    >
      <styled.span
        className="switch__label"
      >
        <span
          className="c1 switch__label"
        >
          Unit:
        </span>
      </styled.span>
      <styled.button
        className="switch__button"
        key="em"
        onClick={[Function]}
        selected={true}
      >
        <button
          className="c2 switch__button"
          onClick={[Function]}
          selected={true}
        >
          em
        </button>
      </styled.button>
      <styled.button
        className="switch__button"
        key="rem"
        onClick={[Function]}
        selected={false}
      >
        <button
          className="c3 switch__button"
          onClick={[Function]}
          selected={false}
        >
          rem
        </button>
      </styled.button>
    </div>
  </styled.div>
</Switch>
`;
